recipes/icu/all/conanfile.py

```diff
@@ -110,6 +117,10 @@ def build_requirements(self):
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        if self.options.dat_package_file:
+            dat_package_file = glob.glob(os.path.join(self.source_folder, self._source_subfolder, "source", "data", "in", "*.dat"))
+            if dat_package_file:
+                shutil.copy(str(self.options.dat_package_file), dat_package_file[0])
```

Comment:
`source` cannot depend on options and settings.
Perhaps we need to add a hook to enforce this..

You need to conditionally copy data in the `build` or `package` method.

Reply:
Moved the copy of the prebuild .dat file to the build method. It can't be done in the package method because this .dat file will be used in the actual building of ICU.